cordova-lib/src/PluginInfo.js

```diff
@@ -288,6 +293,19 @@ function PluginInfo(dirname) {
             return ret;
         });
     };
+
+    self.getFilesAndFrameworks = getFilesAndFrameworks;
+    function getFilesAndFrameworks(platform) {
+        var items = [];
+        // Please avoid changing the order of the calls below, files will be
+        // installed in this order.
+        items = items.concat(self.getSourceFiles(platform));
+        items = items.concat(self.getHeaderFiles(platform));
+        items = items.concat(self.getResourceFiles(platform));
+        items = items.concat(self.getFrameworks(platform));
+        items = items.concat(self.getLibFiles(platform));
```

Comment:
I would recommend :
var items = [];
items = items.concat(self.getSourceFiles(platform),
                                  self.getHeaderFiles(platform),
                                  ... );
// unnecessarily tricky code is a barrier to some, clarity is king